Composer/packages/client/src/App.tsx

```diff
@@ -34,11 +36,21 @@ export const App: React.FC = () => {
     performAppCleanupOnQuit,
     setMachineInfo,
   } = useRecoilValue(dispatcherState);
+  const updateFile = useRecoilCallback((callbackHelpers: CallbackInterface) => async ({ projectId, value }) => {
+    callbackHelpers.set(lgFileState({ projectId, lgFileId: value.id }), value);
+  });
 
   useEffect(() => {
     loadLocale(appLocale);
   }, [appLocale]);
 
+  useEffect(() => {
+    lgWorker.listen(LgEventType.OnUpdateLgFile, (msg) => {
+      const { projectId, payload } = msg.data;
+      updateFile({ projectId, value: payload });
+    });
```

Comment:
This effect will re-run on every re-render. We should make it run only on mount instead, I believe

Also we should remove the listener when the component gets unmounted